src/main/kotlin/com/team4099/robot2023/commands/drivetrain/TargetPoseCommand.kt

```diff
@@ -0,0 +1,156 @@
+        )
+
+    val thetaMaxVel =
+        LoggedTunableValue("Pathfollow/thetaMaxVel", DrivetrainConstants.PID.MAX_AUTO_ANGULAR_VEL)
+    val thetaMaxAccel =
+        LoggedTunableValue("Pathfollow/thetaMaxAccel", DrivetrainConstants.PID.MAX_AUTO_ANGULAR_ACCEL)
+    var desiredAngle: Angle = 0.0.degrees
+    lateinit var odometryCoordinateToTarget: FrameCoordinate.OdometryCoordinate
+
+    init {
+        addRequirements(drivetrain)
+
+        if (RobotBase.isReal()) {
+            thetakP.initDefault(DrivetrainConstants.PID.AUTO_THETA_PID_KP)
+            thetakI.initDefault(DrivetrainConstants.PID.AUTO_THETA_PID_KI)
+            thetakD.initDefault(DrivetrainConstants.PID.AUTO_THETA_PID_KD)
+        } else {
+            thetakP.initDefault(DrivetrainConstants.PID.SIM_AUTO_THETA_PID_KP)
+            thetakI.initDefault(DrivetrainConstants.PID.SIM_AUTO_THETA_PID_KI)
+            thetakD.initDefault(DrivetrainConstants.PID.SIM_AUTO_THETA_PID_KD)
+        }
+
+        thetaPID =
+            ProfiledPIDController(
+                thetakP.get(),
+                thetakI.get(),
+                thetakD.get(),
+                TrapezoidProfile.Constraints(thetaMaxVel.get(), thetaMaxAccel.get())
+            )
+        thetaPID.enableContinuousInput(-PI.radians, PI.radians)
+    }
+
+    override fun initialize() {
+        thetaPID.reset(drivetrain.odomTRobot.rotation)
+
+        odometryCoordinateToTarget = if (targetCoordinate is FrameCoordinate.FieldCoordinate) {
+            FrameCoordinate.OdometryCoordinate(
+                drivetrain.odomTField.asPose2d().transformBy(targetCoordinate.toTransform2d())
+            )
+        } else {
+            targetCoordinate as FrameCoordinate.OdometryCoordinate // Always true, kotlin type casting sucks sometimes
+        }
+    }
+
+    override fun execute() {
+        Logger.recordOutput("ActiveCommands/TargetPoseCommand", true)
+
+        val currentPose = drivetrain.odomTRobot
+        val currentFieldVelocity = fieldVelocitySupplier()
+        val relativeToRobotPose = odometryCoordinateToTarget.toPose2d().relativeTo(currentPose)
+
+        desiredAngle = currentPose.rotation + atan2(relativeToRobotPose.y.inMeters, relativeToRobotPose.x.inMeters).radians
+
+        val thetaFeedback = thetaPID.calculate(currentPose.rotation, desiredAngle)
+
+        drivetrain.currentRequest =
+            Request.DrivetrainRequest.OpenLoop(
+                thetaFeedback,
+                driver.driveSpeedClampedSupplier(driveX, driveY, slowMode),
+                fieldOriented = true
+            )
+
+        Logger.recordOutput("AutoLevel/CurrentYawDegrees", drivetrain.odomTRobot.rotation.inDegrees)
+        Logger.recordOutput("AutoLevel/DesiredYawDegrees", desiredAngle.inDegrees)
+        Logger.recordOutput("AutoLevel/thetaFeedbackDPS", thetaFeedback.inDegreesPerSecond)
+    }
+
+    override fun isFinished(): Boolean {
+        return (drivetrain.odomTRobot.rotation - desiredAngle).absoluteValue <
```

Comment:
i don't think we want isFinished to ever return false? as long as the trigger is pressed down, we should be doing a heading lock.

Reply:
wdym?